inc/global_content/namespace.php

```diff
@@ -61,6 +62,9 @@ function bootstrap_site() : void {
 	// Handle network admin sites list.
 	add_filter( 'manage_sites_action_links', __NAMESPACE__ . '\\sites_list_row_actions', 10, 2 );
 	add_filter( 'display_site_states', __NAMESPACE__ . '\\add_global_site_state', 10, 2 );
+
+	// Handles passing current user cookies to global content requests to allow authenticated requests.
+	add_filter( 'http_request_args', __NAMESPACE__ . '\\filter_global_content_requests_for_auth', 10, 2 );
```

Comment:
Applying this to all requests might cause some weird unintentional behaviour; can we add it to individual requests instead, or is that going to be a gigantic pain?

Reply:
Not really, Rob asked for a global solution so it works with shared blocks and other features in the future.

We can always whitelist requests if needed. 

Unless we have a specific concern in mind about this, I'd say let's keep it and address issues as they come up.

I think the more concerning issue would be the selection of cookies to send, rather than the requests to handle. But again I couldn't think of a reason not to forward everything.

Reply:
Yeah I'd rather this solution "just work" for anyone building on top of the global content repo. This is only going to apply to requests coming from within the same application fwiw.

I agree with Shadi in terms of iterating on it if we do find issues, possibly around user roles and permissions for individuals that differ between sites on the network.

One of my previous attempts to solve this involved automatically adding users making requests to the global site to it with the contributor role by default fwiw.